File: hospital-appointment-frontend/src/api/hospital/checkReport.ts
/**
 * 检查报告 API
 */

import request from '../request'

export interface CheckReport {
  reportID: number
  patientID: number
  doctorID: number
  departmentID: number
  appointmentID?: string
  reportName: string
  reportType: string
  reportContent?: string
  doctorAdvice?: string
  reportStatus: number
  checkDate?: string
  createTime: string
  doctorName?: string
  departmentName?: string
}

export const checkReportAPI = {
  /**
   * 获取患者的检查报告列表
   */
  getByPatient: (patientId: number, reportType?: string) => {
    return request.get<CheckReport[]>(`/hospital/report/patient/${patientId}`, {
      params: reportType ? { reportType } : {}
    })
  },

  /**
   * 获取报告详情
   */
  getById: (id: number) => {
    return request.get<CheckReport>(`/hospital/report/${id}`)
  }
}

File: hospital-appointment-backend/src/main/java/com/hospital/appointment/entity/CheckReport.java
package com.hospital.appointment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("check_report")
public class CheckReport implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long reportID;

    private Long patientID;
    private Long doctorID;
    private Long departmentID;
    private String appointmentID;
    private String reportName;
    private String reportType;
    private String reportContent;
    private String doctorAdvice;
    private Integer reportStatus;
    private LocalDate checkDate;
    private LocalDateTime createTime;

    @TableField(exist = false) private String doctorName;
    @TableField(exist = false) private String departmentName;
}
